src/hera.cpp

```diff
@@ -270,9 +323,6 @@ evmc_result hera_execute(
       );
     }
 
-    heraAssert(hera->engine, "Wasm engine not set.");
-    WasmEngine& engine = *hera->engine;
```

Comment:
Even though it has a terrible overhead, I think `locallyExecuteSystemContract` should create its own engine instance to avoid any memory/state leaking.